src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +220,228 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken)).map(response -> {
+        return response;
+      });
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var sb = new StringBuilder("refreshToken=");
+    sb.append(refreshToken);
+
+    // Path indicates the path that must exist in the requested URL for the client to send the
+    // Cookie header.
+    sb.append("; HttpOnly; Secure; Path=/authn; ");
+
+    // The Max-Age attribute is the number seconds until the token and cookie expires. It also ensures
+    // that the cookie will persist between browser tab sessions.
+    sb.append("Max-Age=");
+    sb.append(ttlSeconds);
+
+    return sb.toString();
+  }
+
+  private String accessTokenCookie(String accessToken, String accessTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(accessTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var sb = new StringBuilder("accessToken=");
+    sb.append(accessToken);
+
+    // Path indicates the path that must exist in the requested URL for the client to send the
+    // Cookie header.
+    sb.append("; HttpOnly; Secure; ");
+
+    // The Max-Age attribute is the number seconds until the token and cookie expires. It also ensures
+    // that the cookie will persist between browser tab sessions.
+    sb.append("Max-Age=");
+    sb.append(ttlSeconds);
+
+    return sb.toString();
+  }
+
+  private Response tokenResponse(JsonObject fetchTokenFuture) {
```

Comment:
fetchTokenFuture is a misleading name because it's a `JsonObject`, not a `Future<JsonObject>`. What about renaming fetchTokenFuture to tokens?

Reply:
Sorry Julian I'm not understanding this one. Can you clarify?

Reply:
Oh hold on, sorry. I was looking at the wrong code. I understand.

Reply:
Fixed